python/paddle/v2/framework/tests/gradient_checker.py

```diff
@@ -301,9 +304,25 @@ def check_grad(self,
 
         check_names = [grad_var_name(name) for name in inputs_to_check]
         for place in places:
-            # get analytical gradients according to different device
-            analytic_grads = self.__get_gradient(forward_op, backward_op,
-                                                 input_vars, check_names, place)
+            # analytic_grads = self.__get_gradient(forward_op, backward_op,
+            #   input_vars, check_names, place)
+            # In fact, the above two lines can be used to replace following
+            # codes. But most of the gradient operators need to handle the case
+            # where one of more of the gradient of the input is not needed.
```

Comment:
`one of more of the gradient of the input is` --> `some input gradients are`

Reply:
Awesome work for making `mul`/`rowwise_add` can ignore some IG.

However, I think the gradient unit-test framework should not be changed for the following reason.

If a user wants to check a `mul` operator gradient without some IGs, he can invoke `check_grad` many times, like `check_grad("mul", input_to_check=["X"], ignore=["Y"])`, `check_grad("mul", input_to_check["Y"], ignore=["X"])`.
   * The disadvantage for invoking `check_grad` many times is the numeric gradient will be calculated many times. It could be slow, but unit tests are not used by users, so it is OK.
   * To make users invoke `check_grad` by themselves can handle if some IGs are not able to be ignored. For example, OpA has two inputs, 'X' and 'Y' and gradient 'Y' should always be calculated, and it `YG` should never be nullptr.
   * To make users invoke `check_grad` can make error log clearer. The error could be `check mul gradient error when calculating X and ignore Y`.

Reply:
Please review https://github.com/qingqing01/Paddle/pull/1

Reply:
@reyoung Thanks！I also thought it is not good to change the `check_grad`  code at first. I merge your patch.